workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactory.kt

```diff
@@ -353,44 +327,32 @@ public fun interface ViewStarter {
  * @see [ScreenViewFactory.forWrapper].
  */
 @WorkflowUiExperimentalApi
-public inline fun <
-  reified WrapperT : Screen,
-  WrappedT : Screen
-  > ScreenViewFactory<WrappedT>.toUnwrappingViewFactory(
-  crossinline unwrap: (wrapperScreen: WrapperT) -> WrappedT
-): ScreenViewFactory<WrapperT> {
-  return toUnwrappingViewFactory(unwrap) { _, wrapperScreen, environment, showUnwrappedScreen ->
-    showUnwrappedScreen(unwrap(wrapperScreen), environment)
-  }
-}
-
-/**
- * Transforms a [ScreenViewFactory] of [WrappedT] into one that can handle instances of [WrapperT].
- *
- * @see [ScreenViewFactory.forWrapper].
- */
-@WorkflowUiExperimentalApi
-public inline fun <
-  reified WrapperT : Screen,
-  WrappedT : Screen
-  > ScreenViewFactory<WrappedT>.toUnwrappingViewFactory(
-  crossinline unwrap: (wrapperScreen: WrapperT) -> WrappedT,
+public inline fun <reified WrapperT, WrappedT> ScreenViewFactory<WrappedT>.toUnwrappingViewFactory(
+  crossinline prepEnvironment: (environment: ViewEnvironment) -> ViewEnvironment = { e -> e },
+  crossinline prepContext: (
+    environment: ViewEnvironment,
+    context: Context
+  ) -> Context = { _, c -> c },
+  crossinline unwrap: (wrapperScreen: WrapperT) -> WrappedT = { it.content },
   crossinline showWrapperScreen: (
     view: View,
     wrapperScreen: WrapperT,
     environment: ViewEnvironment,
     showUnwrappedScreen: (WrappedT, ViewEnvironment) -> Unit
-  ) -> Unit
-): ScreenViewFactory<WrapperT> {
+  ) -> Unit = { _, wrapperScreen, environment, showUnwrappedScreen ->
+    showUnwrappedScreen(wrapperScreen.content, environment)
+  }
+): ScreenViewFactory<WrapperT>
+  where WrapperT : Screen, WrapperT : Wrapper<Screen, WrappedT>, WrappedT : Screen {
```

Comment:
Whoa. This is cool. Has this always been valid syntax?

Reply:
Yup, ancient.